### PR TITLE
Remove RenderedTrajectory and expose an iterator on Position through the interface

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/plugin.cpp [hunk 1]
```diff
@@ -396,29 +396,25 @@ RenderedTrajectory<World> Plugin::RenderedTrajectoryFromIterators(
   }
 
   // Render the trajectory at current time in |World|.
-  DiscreteTrajectory<Navigation>::Iterator initial_it =
-      intermediate_trajectory.Begin();
+  ;
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: O_o

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -396,29 +396,25 @@ RenderedTrajectory<World> Plugin::RenderedTrajectoryFromIterators(
...
-                              final_it.degrees_of_freedom().position()));
-    }
+  for (DiscreteTrajectory<Navigation>::Iterator intermediate_it =
+           intermediate_trajectory.Begin();
+       intermediate_it != intermediate_end;
+       ++intermediate_it) {
+    result.emplace_back(from_navigation_frame_to_world_at_current_time(
+        intermediate_it.degrees_of_freedom().position()));
   }
   VLOG(1) << "Returning a " << result.size() << "-segment trajectory";
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: -point

File: ksp_plugin/plugin.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: The semicolon that wouldn't die.

File: ksp_plugin/plugin.cpp [hunk 3]
```diff
@@ -396,29 +396,25 @@ RenderedTrajectory<World> Plugin::RenderedTrajectoryFromIterators(
...
-  if (initial_it != intermediate_end) {
-    for (auto final_it = initial_it;
-         ++final_it, final_it != intermediate_end;
-         initial_it = final_it) {
-      result.emplace_back(from_navigation_frame_to_world_at_current_time(
-                              initial_it.degrees_of_freedom().position()),
-                          from_navigation_frame_to_world_at_current_time(
-                              final_it.degrees_of_freedom().position()));
-    }
+  for (DiscreteTrajectory<Navigation>::Iterator intermediate_it =
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: `auto intermediate_it = intermediate_trajectory.Begin();`

Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

**Pull request merged**